arlo_server/contests.py

```diff
@@ -0,0 +1,137 @@
+from flask import request, jsonify
+from jsonschema import validate
+from werkzeug.exceptions import BadRequest
+
+from arlo_server import app, db
+from arlo_server.routes import get_election, require_audit_admin_for_organization
+from arlo_server.models import Contest, ContestChoice, Jurisdiction
+
+contest_choice_schema = {
+    "type": "object",
+    "properties": {
+        "id": {"type": "string"},
+        "name": {"type": "string"},
+        "numVotes": {"type": "integer", "minimum": 0},
+    },
+    "required": ["id", "name", "numVotes"],
+}
+
+contest_schema = {
+    "type": "object",
+    "properties": {
+        "id": {"type": "string"},
+        "name": {"type": "string"},
+        "isTargeted": {"type": "boolean"},
+        "choices": {"type": "array", "items": contest_choice_schema},
+        "totalBallotsCast": {"type": "integer", "minimum": 0},
+        "numWinners": {"type": "integer", "minimum": 1},
+        "votesAllowed": {"type": "integer", "minimum": 1},
+        "jurisdictionIds": {"type": "array", "items": {"type": "string"}},
+    },
+    "required": [
+        "id",
+        "name",
+        "isTargeted",
+        "choices",
+        "totalBallotsCast",
+        "numWinners",
+        "votesAllowed",
+        "jurisdictionIds",
+    ],
+}
+
+
+def serialize_contest_choice(db_contest_choice):
+    return {
+        "id": db_contest_choice.id,
+        "name": db_contest_choice.name,
+        "numVotes": db_contest_choice.num_votes,
+    }
+
+
+def serialize_contest(db_contest):
+    return {
+        "id": db_contest.id,
+        "name": db_contest.name,
+        "isTargeted": db_contest.is_targeted,
+        "choices": [serialize_contest_choice(c) for c in db_contest.choices],
+        "totalBallotsCast": db_contest.total_ballots_cast,
+        "numWinners": db_contest.num_winners,
+        "votesAllowed": db_contest.votes_allowed,
+        "jurisdictionIds": [j.id for j in db_contest.jurisdictions],
+    }
+
+
+def deserialize_contest_choice(json_contest_choice, contest_id):
+    return ContestChoice(
+        id=json_contest_choice["id"],
+        contest_id=contest_id,
+        name=json_contest_choice["name"],
+        num_votes=json_contest_choice["numVotes"],
+    )
+
+
+def deserialize_contest(json_contest, election_id):
+    jurisdictions = (
+        db.session.query(Jurisdiction)
+        .filter(Jurisdiction.id.in_(json_contest["jurisdictionIds"]))
```

Comment:
add a filter here that the jurisdiction belong to the `election_id`.

Reply:
added